### PR TITLE
tests: aws support for RELEASE tests

### PR DESCRIPTION
**Pull request by tam7t**
RELEASE=true is a signal from prow that we are running release tests
which exercise the charts/ folder at release time.

**What this PR does / why we need it**: compliment to https://github.com/kubernetes/test-infra/pull/22908

**Which issue(s) this PR fixes** *(optional, using `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when the PR gets merged)*:
Addresses https://github.com/kubernetes-sigs/secrets-store-csi-driver/issues/592

**Special notes for your reviewer**:

### CODE REVIEW

**Review by aramase — COMMENTED** [review]
1. We need a job in test-infra similar to other [presubmit release jobs](https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes-sigs/secrets-store-csi-driver/secrets-store-csi-driver-config.yaml#L310) after this PR is merged.
2. We can skip the make container and docker push steps [here](https://github.com/kubernetes-sigs/secrets-store-csi-driver/blob/master/test/scripts/initialize_eks_cluster.bash#L24) step here if it's `RELEASE`.
3. For `e2e-helm-deploy-release` we don't need a registry because it uses the images from gcr.

File: test/scripts/initialize_eks_cluster.bash [hunk 1]
```diff
@@ -37,4 +39,8 @@ eksctl create iamserviceaccount \
      --approve \
      --region $AWS_REGION
 
-REGISTRY=$ECR_REGISTRY_URL make e2e-helm-deploy
+if [ -z "$RELEASE" ]; then
+    REGISTRY=$ECR_REGISTRY_URL make e2e-helm-deploy
+else
+    make e2e-helm-deploy
```
Comment (aramase) [inline test/scripts/initialize_eks_cluster.bash:45]: This would be `e2e-helm-deploy-release`?

Comment (tam7t) [reply test/scripts/initialize_eks_cluster.bash]: updated to remove installing the driver entirely from this when `RELEASE=true` to be more consistent with the other steps in kubernetes/test-infra#22908

**Review by aramase — COMMENTED** [review]
/lgtm

**Pull request merged**